perf/lua/uri_escape_unescape.lua

```diff
@@ -24,23 +31,37 @@ local decode_sample = uri.escape(encode_sample, escape_opts)
 -- iterations slower than next ones. To avoid such effect we can introduce
 -- "warmup" iterations, disable JIT at all or make JIT compilation
 -- aggressive from the beginning.
---   - hotloop=1 enables compilation in loops on first iteration.
---   - hotexit=1 enables faster compilation for root traces as well as side
---     traces.
+--   - hotloop=1 enables compilation in loops after the first iteration.
+--   - hotexit=1 enables faster compilation for side traces after
+--     exit the parent trace.
 jit.opt.start("hotloop=1", "hotexit=1")
 
-local cycles = 10^3
-local start = clock.monotonic()
-for _ = 1, cycles do
-    uri.escape(encode_sample)
+local CYCLES = 10^3
+
+local tests = {
+    ["uri.escape"] = function()
+        for _ = 1, CYCLES do
+            uri.escape(encode_sample)
+        end
+    end,
+    ["uri.unescape"] = function()
+        for _ = 1, CYCLES do
+            uri.unescape(decode_sample)
+        end
+    end
+}
+
+local function run_test(testname, func)
+    local real_time = clock.time()
+    local cpu_time = clock.proc()
+    func()
+    local real_delta = clock.time() - real_time
+    local cpu_delta = clock.proc() - cpu_time
+    bench:add_result(testname, real_delta, cpu_delta, CYCLES)
 end
-local encode_time = cycles / (clock.monotonic() - start)
 
-start = clock.monotonic()
-for _ = 1, cycles do
-    uri.unescape(decode_sample)
+for testname, test_func in pairs(tests) do
```

Comment:
I'm a bit worrying about an unpredictable order of the subtests.

Reply:
Fixed, changed the tests structure and replaced `pairs()` with `ipairs()`.